apps/analytics/views.py

```diff
@@ -42,3 +50,41 @@ def get_participant_count(request, challenge_pk):
     participant_count = ParticipantCount(participant_count)
     serializer = ParticipantCountSerializer(participant_count)
     return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['GET', ])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail, IsChallengeCreator))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_submission_count(request, challenge_pk, duration):
+    '''
+        Returns submission count for a challenge according to the duration
+        Valid values for duration are all, daily, weekly and monthly.
+    '''
+    # make sure that a valid url is requested.
+    if duration.lower() not in ('all', 'daily', 'weekly', 'monthly'):
+        response_data = {'error': 'Wrong url pattern!'}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    challenge = get_challenge_model(challenge_pk)
+
+    challenge_phase_ids = challenge.challengephase_set.all().values_list('id', flat=True)
+
+    q_params = {'challenge_phase__id__in': challenge_phase_ids}
+    since_date = None
+    if duration.lower() == 'daily':
+        since_date = timezone.now().date()
+
+    elif duration.lower() == 'weekly':
+        since_date = (timezone.now() - timedelta(days=1)).date()
```

Comment:
Shouldn't the timedelta should be `7` instead of `1`? Please let me know if I am wrong.